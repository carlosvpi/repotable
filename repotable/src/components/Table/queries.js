import gql from 'graphql-tag'

export const repoQuery = gql`
query RepoQuery ($name: String!, $owner: String!) {
	repository(name: $name, owner: $owner) {
		name
		description
		forks {
			totalCount
		}
		stargazers {
			totalCount
		}
		url
	}
}
`