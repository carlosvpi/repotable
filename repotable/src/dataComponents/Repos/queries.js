import gql from 'graphql-tag'

export const repoQuery = gql`
query RepoQuery {
	viewer {
		repositories (first: 10) {
			totalCount
			nodes {
				name
				description
				forks {
					totalCount
				}
				stargazers {
					totalCount
				}
				url

			}
		}
	}
}
`