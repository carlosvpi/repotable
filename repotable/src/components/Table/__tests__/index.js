import React from 'react'
import renderer from 'react-test-renderer'
import { MockedProvider } from '@apollo/client/testing'
import { Repotable } from '../'
import { repoQuery } from '../queries' 

const mocks = [
  {
    request: { query: repoQuery },
    result: [
      {
        data: {
          viewer: {
            repositories: {
              totalCount: 2,
              nodes: [
                {
                  name: 'repo 1',
                  description: 'description 1',
                  forks: { totalCount: 1 },
                  stargazers: { totalCount: 5 },
                  url: 'url 1'
                },
                {
                  name: 'repo 2',
                  description: 'description 2',
                  forks: { totalCount: 3 },
                  stargazers: { totalCount: 0 },
                  url: 'url 2'
                }
              ]
            }
          }
        }
      }
    ]
  }
]

test('Repotable matches snapshot', async () => {
  const component = renderer.create(<MockedProvider mocks={ mocks }>
    <Repotable />
  </MockedProvider>)
  expect(component.toJSON().children).toContain('Loading')
  await new Promise(resolve => setTimeout(resolve, 0))
  expect(component.toJSON()).toMatchSnapshot()
})