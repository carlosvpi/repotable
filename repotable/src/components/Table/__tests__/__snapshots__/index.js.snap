// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repotable matches snapshot 1`] = `
Array [
  <table
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          colSpan={1}
          role="columnheader"
        >
          name
        </th>
        <th
          colSpan={1}
          role="columnheader"
        >
          Stars
        </th>
        <th
          colSpan={1}
          role="columnheader"
        >
          Forks
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    />
  </table>,
  <div
    className="pagination"
  >
    <button
      disabled={true}
      onClick={[Function]}
    >
      &lt;&lt;
    </button>
     
    <button
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </button>
     
    <button
      disabled={true}
      onClick={[Function]}
    >
      &gt;
    </button>
     
    <button
      disabled={true}
      onClick={[Function]}
    >
      &gt;&gt;
    </button>
     
    <span>
      Page
       
      <strong>
        1
         of 
        0
      </strong>
       
    </span>
    <span>
      | Go to page:
       
      <input
        defaultValue={1}
        onChange={[Function]}
        style={
          Object {
            "width": "100px",
          }
        }
        type="number"
      />
    </span>
     
    <select
      onChange={[Function]}
      value={10}
    >
      <option
        value={10}
      >
        Show 
        10
      </option>
      <option
        value={20}
      >
        Show 
        20
      </option>
      <option
        value={30}
      >
        Show 
        30
      </option>
      <option
        value={40}
      >
        Show 
        40
      </option>
      <option
        value={50}
      >
        Show 
        50
      </option>
    </select>
  </div>,
]
`;
